project/CodeStone1125/README.md

```diff
@@ -0,0 +1,106 @@
+# 2D Collision Detector
+
+
+## Basic Information
+
+Simulate how different type of shape collide in a 2D map.
+
+* github repository: [https://github.com/CodeStone1125/2DCollisionDetector](https://github.com/CodeStone1125/2DCollisionDetector)
+
+## Problem to Solve
+
+The objective of this system is to calculate scenarios for the most common shapes in
+a 2D map, such as rectangles, circles, and polygons. Before we dive deeper into the 
+topic, let's first discuss the "Axis-Aligned Bounding Box" (AABB) algorithm. It forms
+the basisfor many collision detection systems in 2D graphics and game development due
+to its simplicity and efficiency. Moreover, in the 2D world, AABB can effectively handle 
+approximate collision detection.
+
+However, in many cases, we not only need to deal with hundreds but thousands of collisions
+simultaneously. Iterating through every object can be time-consuming, so we introduce the 
+concept of a 'Quadtree.' This data structure enables efficient space separation and management
+by dividing the space into quadrants. It facilitates quick collision checks by concentrating 
+on potential collision areas, significantly improving performance and reducing unnecessary
+calculations. Additionally, the Quadtree can adapt to various scenarios, ensuring high efficiency.
+
+| ![image](https://github.com/CodeStone1125/nsdhw_23au/assets/72511296/75069f10-594b-40b4-8407-5b047b39980c) | ![image](https://github.com/CodeStone1125/nsdhw_23au/assets/72511296/74d93574-06db-40ad-88f3-a2e1fdcbcd5b) |
+|:-----------------------------------:|:-----------------------------------:|
+| **Quadtree Simpson** |**Quadtree rabbit**|
+
+In summary, the Quadtree plays a vital role in enhancing performance, resulting in a 
+smoother gaming experience. It also provides developers with the flexibility and control
+needed to handle different types of collisions and interactions effectively.
+
+Therefore, in this project, my focus will be on efficiently managing a large number of 
+collision detections using Quadtree optimization.
+
+## Prospective users
+
+* Game Developer: Determine whether something in a game world is attach by a character is very
```

Comment:
There may not be more than a thousand of objects to detect in 2D space.  If having so many objects, it may probably be in 3D space.  If continuing with 2D space, you may choose a different sample application.  Or you may change to use 3D space.

Reply:
Sure, I will change my project.